src/Log/README.md

```diff
@@ -0,0 +1,116 @@
+# Log
+
+The Log package is a [psr/log](https://github.com/php-fig/log) compatible logger implementation using WordPress actions to log messages using [Monolog](https://github.com/Seldaek/monolog) behind the scenes.
+
+As this uses WordPress actions, it doesn't couple any logger to your code, and simply won't fire any log
+messages if no logger is configured.
+
+## Quick Start
+
+Register `Log_Definer.php` and `Log_Subscriber.php` in Square One Core.php. Call the Log Actions below
+in your code.
+
+## Log Action Definition
+
+```php
+/**
+ * @param string $logger_hook Use on of the constants defined in Log_Actions.
+ * @param string $message The message to log.
+ * @param mixed[] $additional_data_to_log Additional data to log. 
+ */
+do_action( string $logger_hook, string $message, array $additional_data_to_log );
+```
+
+## Example Usage
+
+```php
+do_action( \Tribe\Libs\Log\Log_Actions::INFO, 'Starting connection...' );
+
+// ... more logic
+
+if ( $status !== 200 ) {
+    do_action( \Tribe\Libs\Log\Log_Actions::ERROR, 'Invalid status returned from API.', [ 'status' => $status ] );
+}
+
+```
+## Log Action List
+
+```php
+do_action( Log_Actions::EMERGENCY, 'This is an emergency!', [ 'emergency' ] );
+do_action( Log_Actions::ALERT, 'This is an alert.', [ 'alert' ] );
+do_action( Log_Actions::CRITICAL, 'This is a critical!', [ 'critical' ] );
+do_action( Log_Actions::ERROR, 'This is an error.', [ 'error' ] );
+do_action( Log_Actions::WARNING, 'This is a warning.', [ 'warning' ] );
+do_action( Log_Actions::NOTICE, 'This is a notice.', [ 'notice' ] );
+do_action( Log_Actions::INFO, 'This is an info message.', [ 'info' ] );
+do_action( Log_Actions::DEBUG, 'This is a debug message.', [ 'debug' ] );
+```
+
+## Configuration
+
+Logs are written to the `/wp-content` folder, in a file called `square-one-<today's date>.log`. 
+
+> **NOTE** be cautious of what is being written to log files and where they are stored to ensure no sensitive information is leaked if used on a production server.
+
+**Change Log Path and Name**
+
+```php
+add_filter( 'tribe/log/path', static fn ( $path ) => '/server/path/to/my.log' );
+```
+
+**Change the Default Log Level**
+
+The default level is `DEBUG`, you can pass a [LogLevel Constant](https://github.com/php-fig/log/blob/master/src/LogLevel.php) in a define to configure a different level.
+
+in wp-config.php
+
+```php
+define( 'TRIBE_LOG_LEVEL', \Psr\Log\LogLevel::EMERGENCY );
+```
+
+**Change the Log Channel**
+
+The log channel is a descriptive name attached to all log message. The default is `square-one`. Change with:
+
+```php
+add_filter( 'tribe/log/channel', static fn ( $channel ) => 'my-client-name' );
+```
```

Comment:
Can we add an example of the log output here, so there is a reference?

Reply:
Check the tests: https://github.com/moderntribe/tribe-libs/pull/95/files#diff-04c1d0a3d4d173cc5ac5fa14b21773f1209233cea6270d42216e564760165d0cR46

`<channel>.<log level> <your message> [<array of extra data>]`

Reply:
yeah I saw the usage, but I was suggesting a "the result in the log should look like `[timestamp] my-client-name.ERORR {message}`"

Reply:
Got it, I'll add it to the example above this section

Reply:
Added with https://github.com/moderntribe/tribe-libs/pull/95/commits/e867f4043234ae4ba4574c5c6f95a806c342ddee